pkg/controller/atlasproject/ipaccess_list.go

```diff
@@ -20,14 +22,20 @@ import (
 type atlasProjectIPAccessList mongodbatlas.ProjectIPAccessList
 
 func (i atlasProjectIPAccessList) Identifier() interface{} {
+	// hack: Atlas adds the CIDRBlock in case IPAddress is specified in the response.
+	// This breaks the contract (one field per List) and doesn't allow to "merge" lists.
+	// So we ignore the CIDRblock in this case.
+	if i.CIDRBlock != "" && i.IPAddress != "" && strings.HasPrefix(i.CIDRBlock, i.IPAddress) {
```

Comment:
Are you sure this `strings.HasPrefix(i.CIDRBlock, i.IPAddress)` check is correct?

Say we get the following (technically valid) output:
```yaml
"cidrBlock": "192.0.2.0/24",
"ipAddress": "192.0.2.15",
```

The check would fail in this case.

Reply:
yes, this will be correct.
The problem is that when we create the IP Access list with some IP - Atlas sets the CIDRBlock with the mask :32:

we:
```
"ipAddress": "192.0.2.15"
```
atlas response:
```
"ipAddress": "192.0.2.15"
"cidrBlock": "192.0.2.15/32",
```

considering the rule that only one field can be set in the requests I believe that if both fields are set and IPAddress is prefix of cidrBlock - this means that the cidrBlock was generated... 
(unless I'm missing something)

Reply:
Maybe just ignore the CIDR block altogether if IP is set?

Reply:
I see... I was confused by this example from the docs that made it seem like it's possible to have a /24 mask: https://docs.atlas.mongodb.com/reference/api/ip-access-list/add-entries-to-access-list/#example-body

Reply:
ah! I see now! That's a good catch - I think the example is buggy, I'll create a DOCS ticket to fix that (and will also ask someone from the Atlas team to review that)

Reply:
Awesome!

Reply:
the final thought - looking at the examples I think I'll be on a more safe side and will use the logic suggested (no prefix check) as I could be missing something and the CIDR may be calculated magically and can have a mask different from the :32

Thanks for point out 👍